src/main/java/microsoft/exchange/webservices/data/util/DateTimeUtils.java

```diff
@@ -114,4 +115,577 @@ private static Date parseInternal(String value, boolean dateOnly) {
     };
   }
 
+  /**
```

Comment:
hmm since we should avoid using utils in case there is a more object related option maybe this can be refactored to smth like `MicorsoftTimeZone` this could extend a `java.util.TimeZone` and have something like `getName()` your OlsonTimezone could then have 2 constructors one for MicrosoftTimeZone and one for a normal TimeZone which could create a MicrosoftTimeZone Object from it and get the name.
What do you think of this?